### DISCUSSION TITLE
DMA Displaced Moving Average

### DISCUSSION THREAD

**Discussion by 200317314** [ooples/OoplesFinance.StockIndicators#54] [Ideas]
I was wondering if there was a way to add the DMA similar to TDAmeritrade 
https://tlc.thinkorswim.com/center/reference/Tech-Indicators/studies-library/C-D/DMA

To be able to use a negative displacement

**Comment by ooples** [comment]
@200317314 I would need more info such as the formula for calculating this moving average in order to add it to the library

**Reply by 200317314** [reply]
would the TDAmeritrade thinkscript help?

nput price = close;
input length = 28;
input displace = -14;


def dPrice = price[displace];
def isNaNdPrice = IsNaN(dPrice);
def varLength = sum(!isNaNdPrice, length);
def varSum = sum(if isNaNdPrice then 0 else dPrice, length);

plot DMA = if IsNaN(price) or varLength == 0 then Double.NaN else varSum / varLength;
DMA.DefineColor("Displaced", GetColor(0));
DMA.DefineColor("Effective", GetColor(1));
DMA.AssignValueColor(if isNaNdPrice then DMA.color("Effective") else DMA.color("Displaced"));

**Reply by 200317314** [reply]
@ooples I found this might help https://github.com/bizcad/LeanITrend/blob/80afafc8eb317bda81deac203ca8128e6f9c4e78/Algorithm.CSharp/DisplacedMovingAverageRibbon.cs#L31

**Reply by ooples** [reply]
@200317314 I already have a displaced moving average in my library that contains this exact algorithm but I believe I use different parameters by default. My parameters can be customized of course. I believe I just called it Displaced Moving Average

**Reply by 200317314** [reply]
@ooples I checked with the latest nuget packet, I dont see Displaced Moving Average here?
![image](https://github.com/ooples/OoplesFinance.StockIndicators/assets/31968510/c256a809-076c-4156-bf7c-7d671c70f2e9)

**Reply by 200317314** [reply]
@ooples actually i browse the github repo and cant find it, did you maybe not commit this?

**Reply by ooples** [reply]
@200317314 I don't think I used displaced in the name of the method but I have written so many scripts at this point that I'm having trouble remembering which one. I do know that what you want is essentially done in the CalculateAlligatorIndex formula and I would recommend to fork this repo so you can take a look at the code. It would take some customization to change the default parameters but you could use the same length for both the offset amount and the length and change the default moving average to a simple moving average and it will be identical to what you are asking for. I know I have this exact formula in my code somewhere but will let you know when I remember which one it is.

**Reply by 200317314** [reply]
@ooples thank you so much, I appreciate you
